js/bcex.js

```diff
@@ -0,0 +1,466 @@
+        if ('data' in response) {
+            orders = response['data'];
+        }
+        let market = this.markets_by_id[marketId];
+        let results = this.parseOrders (orders, market, since, limit);
+        return results;
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let request = {};
+        let marketId = this.marketId (symbol);
+        request['type'] = 'all';
+        if (typeof symbol !== 'undefined') {
+            request['symbol'] = marketId;
+        }
+        let orders = [];
+        let response = await this.privatePostApiOrderTradeList (this.extend (request, params));
+        if ('data' in response) {
+            orders = response['data'];
+        }
+        let market = this.markets_by_id[marketId];
+        let results = this.parseOrders (orders, market, since, limit);
+        return results;
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        await this.loadMarkets ();
+        let order = {
+            'symbol': this.marketId (symbol),
+            'type': side,
+            'price': price,
+            'number': amount,
+        };
+        let response = await this.privatePostApiOrderCoinTrust (this.extend (order, params));
+        let data = response['data'];
+        return {
+            'info': response,
+            'id': this.safeString (data, 'order_id'),
+        };
+    }
+
+    async cancelOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let request = {};
+        if (typeof symbol !== 'undefined') {
+            request['symbol'] = symbol;
+        }
+        if (typeof id !== 'undefined') {
+            request['order_id'] = id;
+        }
+        let results = await this.privatePostApiOrderCancel (this.extend (request, params));
+        return results;
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let request = '/' + this.implodeParams (path, params);
+        let query = this.omit (params, this.extractParams (path));
+        if (method === 'GET') {
+            if (Object.keys (query).length)
+                request += '?' + this.urlencode (query);
+        }
+        let url = this.urls['api'] + request;
+        if (method === 'POST') {
+            let messageParts = [];
+            let keys = Object.keys (params);
+            let paramsKeys = keys.sort ();
+            for (let i = 0; i < paramsKeys.length; i++) {
+                let paramKey = paramsKeys[i];
+                let param = params[paramKey];
+                messageParts.push (this.encode (paramKey) + '=' + this.encodeURIComponent (param));
+            }
+            if (api === 'private') {
+                this.checkRequiredCredentials ();
+                messageParts.unshift ('api_key=' + this.encodeURIComponent (this.apiKey));
```

Comment:
Unfortunately, `.unshift` isn't transpileable at the moment, we will need to make it plain-c style.